data/src/main/kotlin/com/vultisig/wallet/data/crypto/TonHelper.kt

```diff
@@ -0,0 +1,111 @@
+@file:OptIn(ExperimentalStdlibApi::class)
+
+package com.vultisig.wallet.data.crypto
+
+import com.google.protobuf.ByteString
+import com.vultisig.wallet.data.models.Chain
+import com.vultisig.wallet.data.models.SignedTransactionResult
+import com.vultisig.wallet.data.models.payload.BlockChainSpecific
+import com.vultisig.wallet.data.models.payload.KeysignPayload
+import com.vultisig.wallet.data.tss.getSignature
+import com.vultisig.wallet.data.utils.Numeric
+import tss.KeysignResponse
+import wallet.core.jni.AnyAddress
+import wallet.core.jni.CoinType
+import wallet.core.jni.DataVector
+import wallet.core.jni.PublicKey
+import wallet.core.jni.PublicKeyType
+import wallet.core.jni.TransactionCompiler
+import wallet.core.jni.proto.TheOpenNetwork
+
+object TonHelper {
+
+    private fun getPreSignedInputData(payload: KeysignPayload): ByteArray {
+        require(payload.coin.chain == Chain.Ton) { "Coin is not TON" }
+
+        val (sequenceNumber, expireAt, bounceable) = payload.blockChainSpecific as? BlockChainSpecific.Ton
+            ?: throw RuntimeException("Fail to get Ton chain specific")
+
+        val toAddress = AnyAddress(payload.toAddress, CoinType.TON)
+
+        val publicKey =
+            PublicKey(payload.coin.hexPublicKey.hexToByteArray(), PublicKeyType.ED25519)
+
+        val transfer = TheOpenNetwork.Transfer.newBuilder()
+            .setDest(toAddress.description())
+            .setAmount(payload.toAmount.toLong())
+            .setMode(
+                TheOpenNetwork.SendMode.PAY_FEES_SEPARATELY_VALUE
+                        or TheOpenNetwork.SendMode.IGNORE_ACTION_PHASE_ERRORS_VALUE
+            )
+            .setBounceable(false) // TODO
```

Comment:
On IOS , it check the memo , if it is `d` or `w` , then it set bounceable to `true`